lib/logstash/devutils/rspec/logstash_helpers.rb

```diff
@@ -36,32 +37,45 @@ def sample(sample_event, &block)
         pipeline.instance_eval { @filters.each(&:register) }
 
         event.each do |e|
-          pipeline.filter(e) {|new_event| results << new_event }
+          pipeline.filter(e) {|new_event| results << new_event}
         end
 
         pipeline.flush_filters(:final => true) do |e|
           results << e unless e.cancelled?
         end
 
-        results
+        results.select{|e| !e.cancelled?}
```

Comment:
Why do we need this filter in here? aren't we doing the same work just before in the flush filter call?